frontends/infinite-corridor/src/api/learning-resources/favorites.ts

```diff
@@ -0,0 +1,60 @@
+import { LearningResource, PaginatedListItems } from "ol-search-ui"
+import { PaginationSearchParams } from "ol-util"
+import axios from "../../libs/axios"
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query"
+import { urls, keys } from "./urls"
+import { modifyCachedSearchResource } from "./search"
+import { invalidateResourceQueries } from "./util"
+
+const useFavoritesListing = (options?: PaginationSearchParams) => {
+  const url = urls.favorite.listing(options)
+  const key = keys.favorites.listing.page(options)
+  return useQuery<PaginatedListItems>(key, () =>
+    axios.get(url).then(res => res.data)
+  )
+}
+
+const useFavorite = () => {
+  const queryClient = useQueryClient()
+  return useMutation({
+    mutationFn: (resource: LearningResource) => {
+      const url = urls.resource.favorite(resource.object_type, resource.id)
+      return axios.post(url).then(res => res.data)
+    },
+    onSuccess(_data, resource) {
+      queryClient.invalidateQueries({ queryKey: keys.favorites.all })
+      invalidateResourceQueries(queryClient, resource)
```

Comment:
cf https://github.com/mitodl/open-discussions/pull/3997/files#diff-5372eb9a81a96ae31f18aaf3ee85f8d67573633f8733417b6095fe7b053b2ae5L130

The hooks in this PR did not change, except now they use [`invalidateResourceQueries`](https://github.com/mitodl/open-discussions/pull/3997/files#diff-3a854f99846c5671b7f8f3c74edc218520c5ddd4fa08614d6e6388764d259ce4R30) to invalidate